content/sips/sip-311.md

```diff
@@ -88,14 +109,46 @@ _Note that this function must conform to the existing CCIP interface._
 
 <!--Test cases for an implementation are mandatory for SIPs but can be included with the implementation.-->
 
-Relevant tests will be developed during implementation.
+Relevant tests will be developed during implementation, including but not limited to:
+
+#### Unit Tests
+
+Two new unit tests are added to test the new contracts. These tests will mock the required external interfaces (i.e. `CcipRouter` and `CcipTokenPool`).
+
+For `USDTokenModule`:
+
+- It reverts transactions coming from an unauthorized address (i.e. non-CCIP address).
+- It reverts if the sender does not have enough snxUSD balance or allowance.
+- It gets deployed and registered correctly.
+- It applies the expected USD parameters.
+
+For `CcipReceiverModule`:
+
+- It reverts if caller is not the `CcipRouter`.
+- It reverts if chain is not supported.
+- It reverts if message sender on other chain is not self.
+- It forwards message to specified caller.
+
+#### Integration Tests
+
+One integration test to cover the initiation of a cross-chain transfer.
+
+- Verify it burns the correct amount of snxUSD on the source chain.
```

Comment:
This can be replaced by verifying the allowance that the Router has been given via approve to transferFrom the amount of snxUSD transferred.